### PR TITLE
pipenv

### PR DESCRIPTION
**Pull request by hshaosf**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -32,12 +32,13 @@ jobs:
           command: |
             python3 -m venv venv
```
Comment (josh-chou) [inline .circleci/config.yml]: Don't really need venv anymore since pipenv creates a virtualenv also.

File: .circleci/config.yml [hunk 2]
```diff
@@ -32,12 +32,13 @@ jobs:
...
+            pipenv install
 
       - save_cache:
           paths:
             - ./venv
-          key: v1-dependencies-{{ checksum "requirements.txt" }}
+          key: v1-dependencies-{{ checksum "Pipfile.lock" }}
       
       - run:
           name: start gunicorn
```
Comment (josh-chou) [inline .circleci/config.yml:42]: pipenv run gunicorn 'service.microservice:start_service()'

Comment (josh-chou) [reply .circleci/config.yml:42]: same with tests.
pipenv run python -m pytest --cov=service tests/  --cov-fail-under=100
I couldn't get coveralls to work with pipenv.  Might need to run it as a compound command along with pytest.

File: .circleci/config.yml (hunk 1 above)
Comment (hshaosf) [reply .circleci/config.yml]: I was mostly just focusing on requirements.txt in the first pass to use pipenv. commit 2ca1926 now contains the latest.

File: .circleci/config.yml (hunk 2 above)
Comment (hshaosf) [reply .circleci/config.yml:42]: similar to above: 2ca1926 
re coveralls: used `pipenv run coveralls`

**Review by josh-chou — APPROVED** [review]
👍

**Pull request merged**